### ISSUE TITLE
DDC_DeepCoral中的mmd计算

### ISSUE THREAD

**Issue by A4ce** [jindongwang/transferlearning#204]
王博你好，我想请教一下，为什么计算transfer_loss特别使用不计算梯度的方法，计算出来的transfer_loss只是一个常数。
with torch.no_grad():
    XX = torch.mean(kernels[:batch_size, :batch_size])
    YY = torch.mean(kernels[batch_size:, batch_size:])
    XY = torch.mean(kernels[:batch_size, batch_size:])
    YX = torch.mean(kernels[batch_size:, :batch_size])
    loss = torch.mean(XX + YY - XY - YX)

**Comment by jindongwang** [comment]
不好意思我没明白你的问题。。能详细说一下吗？

**Comment by A4ce** [comment]
DDC_DeepCoral的mmd.py文件夹里面，计算迁移损失中疑惑处如下
![image](https://user-images.githubusercontent.com/49636417/102986514-e910fb00-454b-11eb-87fb-83bce2a5a5fd.png)
计算迁移损失特地使用不计算梯度的方式是有什么原因吗？
得到的transfer_loss是一个常数，应该不能通过计算反向传播求导。
![image](https://user-images.githubusercontent.com/49636417/102986598-11005e80-454c-11eb-8d35-bdb6626b3401.png)

**Comment by A4ce** [comment]
王博，你好。DDC_DeepCoral的mmd.py文件夹里面，计算迁移损失中疑惑处如下
计算迁移损失特地使用不计算梯度的方式是有什么原因吗？
得到的transfer_loss
是一个常数，应该不能通过pytorch反向传播求导。













在 2020-12-23 17:24:35，"Jindong Wang" <notifications@github.com> 写道：

不好意思我没明白你的问题。。能详细说一下吗？

**Comment by jindongwang** [comment]
明白了。MMD计算一直是常数的，我这样写是当时做一个项目，显存不够了，后来重新修改代码的时候忘记了。。。实际不应该加with torch.no_grad的！我这就改了，谢谢你指出来。

**Comment by A4ce** [comment]
谢谢，解决了我的疑问。
